### PR TITLE
HADOOP-8728. Display (fs -text) shouldn't hard-depend on Writable serialized sequence files.

### PR DESCRIPTION
**Pull request by hotcodemacha**
### Description of PR
Display (fs -text) shouldn't hard-depend on Writable serialized sequence files.

The Display command (fs -text) currently reads only Writable-based SequenceFiles. This isn't necessary to do, and prevents reading non-Writable-based serialization in SequenceFiles from the shell.

JIRA - HADOOP-8728

### How was this patch tested?
UT

### For code changes:

- [X] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 43s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  40m 41s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  24m 54s |  |  trunk passed with JDK Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  22m 10s |  |  trunk passed with JDK Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   1m 59s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m  4s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   1m 30s |  |  trunk passed with JDK Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m  5s |  |  trunk passed with JDK Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   3m  6s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  24m 57s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m  8s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  24m 58s |  |  the patch passed with JDK Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  24m 58s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  22m 55s |  |  the patch passed with JDK Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |  22m 55s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   1m 26s | [/results-checkstyle-hadoop-common-project_hadoop-common.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5010/1/artifact/out/results-checkstyle-hadoop-common-project_hadoop-common.txt) |  hadoop-common-project/hadoop-common: The patch generated 2 new + 139 unchanged - 12 fixed = 141 total (was 151)  |
| +1 :green_heart: |  mvnsite  |   1m 52s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 13s |  |  the patch passed with JDK Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   0m 58s |  |  the patch passed with JDK Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   2m 50s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  23m 50s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  19m 21s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 22s |  |  The patch does not generate ASF License warnings.  |
|  |   | 225m 43s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5010/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5010 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 89a0146c9c8b 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / a1b61e136d05ad4069f96c625470c35340302375 |
| Default Java | Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5010/1/testReport/ |
| Max. process+thread count | 2075 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5010/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/shell/Display.java [hunk 1]
```diff
@@ -217,8 +216,8 @@ protected void processPath(PathData item) throws IOException {
 
   protected class TextRecordInputStream extends InputStream {
     SequenceFile.Reader r;
-    Writable key;
-    Writable val;
+    Object key;
```
Comment (goiri) [inline hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/shell/Display.java:219]: What is the use case to not base it on Writable? The whole principal is to use Writable.

Comment (hotcodemacha) [reply hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/shell/Display.java:219]: Hi @goiri. Thanks for your review. 

Currently - The Display command (fs -text) currently reads only Writable-based SequenceFiles. This isn't necessary to do, and prevents reading non-Writable-based serialization in SequenceFiles from the shell.

**Comment by hotcodemacha** [comment]
@aajisaka @goiri please help in reviewing this PR. Thanks

**Review by aajisaka — APPROVED** [review]
LGTM.

**Comment by aajisaka** [comment]
Thank you @ashutoshcipher for your contribution.

**Comment by hotcodemacha** [comment]
Thanks @aajisaka for reviewing and merging.

**Pull request merged**